base/MAPL_ObsUtil.F90

```diff
@@ -0,0 +1,544 @@
+       end if
+
+
+    _RETURN(_SUCCESS)
+
+  end function get_filename_from_template_use_index
+
+
+
+  subroutine get_var_from_name_w_group (var_name, var2d, filename, rc)
+    character(len=ESMF_MAXSTR), intent(in) :: var_name, filename
+    real(ESMF_KIND_R8), intent(inout) :: var2d(:,:)
+    integer, optional, intent(out) :: rc
+
+    integer :: i, j
+    character(len=ESMF_MAXSTR) :: grp1, grp2
+    character(len=ESMF_MAXSTR) :: short_name    
+    integer :: ncid, ncid2, varid
+    logical :: found_group
+    integer :: status
+
+
+    i=index(var_name, '/')
+    if (i>0) then
+       found_group = .true.
+       grp1 = var_name(1:i-1)
+       j=index(var_name(i+1:), '/')
+       if (j>0) then
+          grp2=var_name(i+1:i+j-1)
+          short_name=var_name(i+j+1:)
+       else
+          grp2=''
+          short_name=var_name(i+1:)             
+       endif
+       i=i+j
+    else
+       found_group = .false.
+       grp1 = ''
+       grp2=''
+       short_name=var_name
+    endif
+
+    call check_nc_status(nf90_open(filename, NF90_NOWRITE, ncid2), _RC)
+    if ( found_group ) then
+       call check_nc_status(nf90_inq_ncid(ncid2, grp1, ncid), _RC)
+       if (j>0) then
+          call check_nc_status(nf90_inq_ncid(ncid, grp2, ncid2), _RC)
+          ncid=ncid2
+       endif
+    else
+       print*, 'no grp name'
+       ncid=ncid2
+    endif
+    call check_nc_status(nf90_inq_varid(ncid, short_name, varid), _RC)
+    call check_nc_status(nf90_get_var(ncid, varid, var2d), _RC)
+!!    call check_nc_status(nf90_close(ncid), _RC)
+
+    _RETURN(_SUCCESS)
+
+  end subroutine get_var_from_name_w_group
+
+
+  subroutine sort_three_arrays_by_time(U,V,T,rc)
+    use MAPL_SortMod
+    real(ESMF_KIND_R8), intent(inout) :: U(:), V(:), T(:)
+    integer, optional, intent(out) :: rc
+
+    integer :: i, len
+    integer, allocatable :: IA(:)
+    integer(ESMF_KIND_I8), allocatable :: IX(:)
+    real(ESMF_KIND_R8), allocatable :: X(:)
+
+    _ASSERT (size(U)==size(V), 'U,V different dimension')
+    _ASSERT (size(U)==size(T), 'U,T different dimension')
+    len = size (T)
+
+    allocate (IA(len), IX(len), X(len))
+    do i=1, len
+       IX(i)=T(i)
+       IA(i)=i
+    enddo
+    call MAPL_Sort(IX,IA)
+
+    X = U
+    do i=1, len
+       U(i) = X(IA(i))
+    enddo
+    X = V
+    do i=1, len
+       V(i) = X(IA(i))
+    enddo
+    X = T
+    do i=1, len
+       T(i) = X(IA(i))
+    enddo
+    _RETURN(_SUCCESS)
+  end subroutine sort_three_arrays_by_time
+
+
+  subroutine sort_four_arrays_by_time(U,V,T,ID,rc)
+    use MAPL_SortMod
+    real(ESMF_KIND_R8) :: U(:), V(:), T(:)
+    integer :: ID(:)
+    integer, optional, intent(out) :: rc
+
+    integer :: i, len
+    integer, allocatable :: IA(:)
+    integer(ESMF_KIND_I8), allocatable :: IX(:)
+    real(ESMF_KIND_R8), allocatable :: X(:)
+    integer, allocatable :: NX(:)
+
+    _ASSERT (size(U)==size(V), 'U,V different dimension')
+    _ASSERT (size(U)==size(T), 'U,T different dimension')
```

Comment:
I'm a little bit surprised that CPP works with the extra space after `_ASSERT`.   Learned something.  Nonetheless, I recommend following the usual style without the space.   No need to fix for now though.

Reply:
I see what you mean.  Google the CPP style shows something, like, the space between a function or array name and the left parenthesis works with compiler, but is/may not be a preferred style.  I have changed it. Thanks.